File: C++/102. Binary Tree Level Order Traversal.cpp
#include <bits/stdc++.h>

using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

vector<vector<int>> levelOrder(TreeNode *root)
{
    vector<vector<int>> result;

    vector<int> level;
    int level_len;

    queue<TreeNode *> que;

    if (root)
        que.push(root);

    while (!que.empty())
    {
        level_len = que.size();

        while (level_len--)
        {
            root = que.front();
            que.pop();

            level.push_back(root->val);

            if (root->left)
                que.push(root->left);

            if (root->right)
                que.push(root->right);
        }
        result.push_back(level);
        level.clear();
    }

    return result;
}

File: Java/102. Binary Tree Level Order Traversal.java

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();

        if (root == null)
            return result;

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            int size = queue.size();
            List<Integer> levels = new ArrayList<>(size);

            while (size-- > 0) {
                root = queue.poll();
                levels.add(root.val);

                if (root.left != null)
                    queue.add(root.left);

                if (root.right != null)
                    queue.add(root.right);
            }
            result.add(levels);
        }

        return result;
    }

    /* ignore-boiler-code */
    public class TreeNode {
        public int val;
        public TreeNode left;
        public TreeNode right;

        public TreeNode() {
        }

        public TreeNode(int val) {
            this.val = val;
        }

        public TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }
}